doc/create-announcement-article.rb

```diff
@@ -0,0 +1,336 @@
+  If we specify as below, "*<" and "*>" work as &&.
+
+    'content @ "Groonga" *< content @ "Mroonga"'
+
+* Groonga may not return records that should match caused by "GRN_II_CURSOR_SET_MIN_ENABLE".
+    ARTICLE
+  end
+end
+
+class FacebookJaArticleGenerator < FacebookArticleGenerator
+  def initialize(release_date, groonga_version, groonga_previous_version, groonga_org_repository)
+    super(release_date, groonga_version, groonga_previous_version, groonga_org_repository)
+    @input_file_path = "./locale/ja/text/news.txt"
+    @output_file_path = "./tmp/facebook-ja-#{release_date}-groonga-#{groonga_version}.txt"
+    @release_headline_regexp_pattern = ".*リリース.+\\d\\d\\d\\d-\\d\\d-\\d\\d.*\\n=.+"
+
+  end
+
+  def generate_article
+    <<-"ARTICLE"
+Groonga #{@groonga_version} をリリースしました！
+
+    https://groonga.org/ja/docs/news.html#release-#{@groonga_version_in_link}
+
+変更点一覧:
+
+    https://groonga.org/ja/blog/#{@release_date_in_link}/groonga-#{@groonga_version_in_link}.html
+
+変更内容
+========
+
+主な変更点は以下の通りです。
+
+#{extract_latest_release_note}
+
+既知の問題
+----------
+
+* 現在Groongaには、ベクターカラムに対してデータを大量に追加、削除、更新した際にデータが破損することがある問題があります。
+
+* "*<" と "*>" は、filter条件の右辺に "query()" を使う時のみ有効です。もし、以下のように指定した場合、 "*<" と "*>" は "&&" として機能します。
+
+  'content @ "Groonga" *< content @ "Mroonga"'
+
+* "GRN_II_CURSOR_SET_MIN_ENABLE" が原因でマッチするはずのレコードを返さないことがあります。
+    ARTICLE
+  end
+end
+
+class TwitterEnArticleBaseGenerator < ArticleGenerator
+  def initialize(release_date, groonga_version, groonga_previous_version, groonga_org_repository)
+    super(release_date, groonga_version, groonga_previous_version, groonga_org_repository)
+    @output_file_path = "./tmp/twitter-en-#{release_date}-groonga-#{groonga_version}-base.txt"
+  end
+
+  def generate_article
+    "Groonga #{@groonga_version} has been released!(#{@release_date}) " + 
+    "https://groonga.org/en/blog/#{@release_date_in_link}/groonga-#{@groonga_version}.html"
+  end
+end
+
+class TwitterJaArticleBaseGenerator < ArticleGenerator
+  def initialize(release_date, groonga_version, groonga_previous_version, groonga_org_repository)
+    super(release_date, groonga_version, groonga_previous_version, groonga_org_repository)
+    @output_file_path = "./tmp/twitter-ja-#{release_date}-groonga-#{groonga_version}-base.txt"
+  end
+
+  def generate_article
+    "Groonga #{@groonga_version}をリリースしました！(#{@release_date}) " +
+    "https://groonga.org/ja/blog/#{@release_date_in_link}/groonga-#{@groonga_version}.html"
+  end
+end
+
+option = {}
+OptionParser.new do |opt|
+  opt.on('--release-date VALUE', "YYYY-MM-DD") { |v| option[:release_date] = v }
+  opt.on('--groonga_version VALUE', "e.g. 12.1.1") { |v| option[:groonga_version] = v }
+  opt.on('--groonga_previous_version VALUE', "e.g. 12.1.0") { |v| option[:groonga_previous_version] = v }
+  opt.on('--groonga_org_repository VALUE', "e.g. $HOME/work/groonga.org") { |v| option[:groonga_org_repository] = v }
+  opt.parse!(ARGV)
+end
+
+generator_class_list = [
```

Comment:
Ruby doesn't have `List` and this is `Array`.

In general, we don't need to add variable value type (`list`, `array` and so on) to variable name because it's not important. (It's only important that "this is a container".) If the variable value is a container, we can just use plural: 

```suggestion
generator_classes = [
```